Feature 291 add black formatting as ci

deniztepe opened the pull request:
The pre-commit hook is activated, after pip installing the `pre-commit` package and running `pre-commit install` on console. You can test it on the feature branch, if you want to. I also added a short documentation of the feature.

deniztepe committed:
Install and configure black as pre-commit #291

deniztepe committed:
Add documentation for black pre-commit #291

deniztepe committed:
Add pre-commit package to extras_require #291

* Update documentation
* Apply black to setup.py

deniztepe committed:
Update changelog #291

FlorianK13 committed:
Update documentation #312

This commit is needed to start the CI pipeline and run the tests.

FlorianK13 committed:
Make flake8 and black compatible #291

Taken from
https://black.readthedocs.io/en/stable/guides/
using_black_with_other_tools.html

FlorianK13 commented:
I have adapted the flake8 file to be black compatible. There is still the pylint file (.pylintrc) in the main directory which is also for code formatting. Can we delete it since we only use black + flake8 ?

deniztepe commented:
I would say, yes?

FlorianK13 committed:
Delete Pylint file #291

Pylint is not really used anymore.

FlorianK13 committed:
Change line length #291

Line length was previously set to 100.
Even though black compability is better with line_length = 88
the code was written with line_length=100 and many
changes would be necessary.

FlorianK13 commented on .pre-commit-config.yaml:
```diff
@@ -0,0 +1,6 @@
+    rev: 22.6.0
+    hooks:
+      - id: black
+        language_version: python3.9
```
what does python3.9 mean here? Can it have conflicts with people using other versions?

deniztepe replied:
It was on the default template. As far as I understood, it determines the python version for the hook. Black is called with this version. I assume it should be compatible with other versions such as 3.7...3-10.

FlorianK13 replied:
Let's merge it then and have a bugfix if it does not work.
